tinyDA/chain.py

```diff
@@ -397,7 +441,7 @@ def _sample_coarse(self):
                 self.accepted_coarse.append(True)
                 self.is_coarse.append(True)
             else:
-                self.chain_coarse.append(self.chain_coarse[-1])
+                self.chain_coarse.append(self.promoted_coarse[-1]) # adapted to use promoted_coarse
```

Comment:
I don't think this is correct. The coarse should always add its own previous link if the proposal is rejected.

Reply:
So, it should be elf.chain_coarse.append(self.chain_coarse[-(self.subchain_length+1)]), right?

Reply:
No, for running the coarse chain, if the sample is rejected, it should always just be `self.chain_coarse[-1]`.

If you would do `self.chain_coarse[-(self.subchain_length+1)]`, then if there had been an accepted sample before, but we now reject, then that previously accepted sample would be disregarded.

Reply:
Oooh, I see. Sorry, I got confused!